sycl/test/esimd/esimd_verify_local_names.cpp

```diff
@@ -0,0 +1,45 @@
+// RUN: %clangxx -fsycl -fsycl-device-only -S %s -o /dev/null
```

Comment:
I'd rather made this an E2E self-checking test (at intel/llvm-test-suite) to ensure that it not only compiles, but there are no surprises during runtime

Reply:
Sure, I can do that. Should I keep this one just to catch anything during lit testing?

Reply:
intel-test-suite PR: https://github.com/intel/llvm-test-suite/pull/1511